### DISCUSSION TITLE
not able to use all hardware capacity

### DISCUSSION THREAD

**Discussion by emrahnazif** [directus/directus#15877] [Archive]
Hi all,

I deployed directus locally and serving it via pm2 in cluster mode.
the hardware is mac mini m1 with 16gb ram. 

Sometimes I do batch processes (I say batch but I do it with individual requests). when I would like to modify 1m items, what I do is,  I create a list of item IDs. Then in python I send requests with ThreadPool with pool size 5 (5 requests in paralel). I can modify lets say 30/40 items in sec. When I monitor directus CPU usage it is very low, around 1%, when I monitor mysql CPU usage, it is not high as well (15-50%).
When I increase the pool size lets say 15-20. I got even slower actions/sec.
Meanwhile, when I do some search actions etc, CPU can hit to 100% (mostly mysql usage).

When I download files via API I can reach over 100 files/secs. It is almost equals to my bandwidth, so it is good. 

When I edit items, I would like to hit the maximum requests/sec that the hardware is capable. Any idea, what am I doing wrong? Or how can I improve the rate?

**Comment by br41nslug** [comment]
The only real limit i know nodejs imposes is memory which you can change (https://blog.appsignal.com/2021/12/08/nodejs-memory-limits-what-you-should-know.html). If CPU is only at 1% that probably indicates there is some other bottleneck blocking the performance like memory speed/memory size/disk speed/network speed.
